ext/storage_sources/azure_store/azure_log_system.h

```diff
@@ -0,0 +1,79 @@
+/*-
+ * Public Domain 2014-present MongoDB, Inc.
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#pragma once
+#include <wiredtiger.h>
+#include <wiredtiger_ext.h>
+
+#include <azure/core/diagnostics/logger.hpp>
+
+#include <atomic>
+#include <map>
+
+const std::map<int32_t, Azure::Core::Diagnostics::Logger::Level> wt_to_azure_verbosity_mapping = {
+  {WT_VERBOSE_ERROR, Azure::Core::Diagnostics::Logger::Level::Error},
+  {WT_VERBOSE_WARNING, Azure::Core::Diagnostics::Logger::Level::Warning},
+  {WT_VERBOSE_INFO, Azure::Core::Diagnostics::Logger::Level::Informational},
+  {WT_VERBOSE_DEBUG_1, Azure::Core::Diagnostics::Logger::Level::Verbose},
+  {WT_VERBOSE_DEBUG_2, Azure::Core::Diagnostics::Logger::Level::Verbose},
+  {WT_VERBOSE_DEBUG_3, Azure::Core::Diagnostics::Logger::Level::Verbose},
+  {WT_VERBOSE_DEBUG_4, Azure::Core::Diagnostics::Logger::Level::Verbose},
+  {WT_VERBOSE_DEBUG_5, Azure::Core::Diagnostics::Logger::Level::Verbose},
+};
+
+const std::map<Azure::Core::Diagnostics::Logger::Level, int32_t> azure_to_wt_verbosity_mapping = {
+  {Azure::Core::Diagnostics::Logger::Level::Error, WT_VERBOSE_ERROR},
+  {Azure::Core::Diagnostics::Logger::Level::Warning, WT_VERBOSE_WARNING},
+  {Azure::Core::Diagnostics::Logger::Level::Informational, WT_VERBOSE_INFO},
+  {Azure::Core::Diagnostics::Logger::Level::Verbose, WT_VERBOSE_DEBUG_5},
```

Comment:
Not sure about `DEBUG_5` here, what sort of testing did you do with the different levels? Is it super noisy at (say) `DEBUG_1`?

Reply:
i havent actually been able to test this so i've just given it the lowest priority

Reply:
OK, that's not unreasonable - especially given that the config for it lives outside the "main" verbose config options.